src/node.rs

```diff
@@ -139,13 +114,13 @@ impl NodeBuilder {
                         this is not supported",
                        full_id.public_id().name());
 
-                let _ = event_sender.send(Event::Terminate);
+                // FIXME(michael): send this event?
```

Comment:
I think we have to return an `Evented<RoutingActionSender, StateMachine>` from `StateMachine::new`, and update the calls to this function in `Node` and `Client` so that they push them on their internal buffer (or send them on the `event_sender` in the case of the non-test Client).